src/app/components/publiclists/challengelist/challengelist.component.ts

```diff
@@ -119,15 +132,38 @@ export class ChallengelistComponent implements OnInit {
     if (this.authService.isLoggedIn()) {
       this.isLoggedIn = true;
     }
+
+    this.authServiceSubscription = this.authService.change.subscribe((authState) => {
+      this.isLoggedIn = authState.isLoggedIn;
+      if (!authState.isLoggedIn && this.router.url === '/challenges/me') {
+        this.router.navigate(['/auth/login']);
+      }
+    });
+
     if (this.router.url === '/challenges/all') {
       this.fetchChallenges();
     } else if (this.router.url === '/challenges/me' && this.authService.isLoggedIn()) {
       this.fetchMyTeams();
     }
+
     this.authServicePublic = this.authService;
     this.routerPublic = this.router;
   }
 
+
+  /**
+   * Listener for page scroll event
+   */
+  @HostListener('window:scroll', [])
+  onWindowScroll(): void {
+    const RECT =  this.document.getElementById('ongoing-challenges').getBoundingClientRect();
+    if (RECT.top < 0) {
```

Comment:
Simplify it like this, `this.isScrollbtnVisible = RECT.top < 0;`

Reply:
done